### PR TITLE
adding ignored keys support for masking metadata keys

### PR DESCRIPTION
**Pull request by enoodle**
Adding support for ignoredKeys

- [x] UT
- [x] example progranm
- [x] REAME

### CODE REVIEW

File: tracer/mask_ignored_keys.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+func arrayToHitMap(arr []string) map[string]bool {
+	hitMap := make(map[string]bool)
+	for _, k := range arr {
+		hitMap[k] = true
+	}
+	return hitMap
+}
+
+func maskNestedJSONKeys(decodedJSON interface{}, ignoredKeysMap map[string]bool) interface{} {
```
Comment (maorlx) [inline tracer/mask_ignored_keys.go:20]: I would change this function to return also if any value has changed (so will return interface{}, bool) - so we will change values (and the final metadata value) only if needed

Comment (enoodle) [reply tracer/mask_ignored_keys.go:20]: how does it matter? it seems more complicated this way (adding many tests for changes across the code) and adds no benefit as assigning values takes negligible amount of time over testing and branching..

Comment (maorlx) [reply tracer/mask_ignored_keys.go:20]: you are right its makes the code more complicated (thats why I was wondering about it)
Do you think it still worth as we json.Marshal every time (per event/metadata key)? not sure its its negligible
if so then lets keep it that way

Comment (enoodle) [reply tracer/mask_ignored_keys.go]: It doesn't but it takes less time to add this then to explain

Comment (enoodle) [reply tracer/mask_ignored_keys.go]: I have added this to the PR

**Review by maorlx — DISMISSED** [review]

**Review by maorlx — APPROVED** [review]

**Pull request merged**